src/main/java/io/github/fileanalysissuite/adaptersdk/impls/jaxrs/AdapterApiImpl.java

```diff
@@ -37,36 +38,49 @@ public AdapterApiImpl(final RepositoryAdapter repositoryAdapter)
 
     @Nonnull
     @Override
-    public AdapterDescriptor adapterDescriptorGet()
+    public Response adapterDescriptorGet()
     {
-        return AdapterDescriptorFunctions.convertToModel(repositoryAdapter.createDescriptor());
+        try {
+            final AdapterDescriptor descriptor = AdapterDescriptorFunctions.convertToModel(repositoryAdapter.createDescriptor());
+            return Response.ok(descriptor).build();
+        } catch (final InterruptedException ex) {
+            return Response.status(Response.Status.SERVICE_UNAVAILABLE).build();
```

Comment:
Since we are not propagating the exception it may not be obvious why 503 is being returned.  Therefore we need to add some logging here to explain that the call has been interrupted.